src/server/ApiRoutes/Search.js

```diff
@@ -68,34 +69,48 @@ function fetchResults(searchKeywords = '', page, sortBy, order, field, filters,
       .catch(console.error))
     .then((response) => {
       const [results, aggregations, drbbResults] = response;
-      return Promise.all(results.itemListElement.map((resultObj) => {
+      const locationCodes = new Set();
+      const { itemListElement } = results;
+      itemListElement.forEach((resultObj) => {
+        let numItemsProcessed = 0;
         const { result } = resultObj;
-        return addLocationUrls(result).then((updatedResult) => {
-          const { holdings } = updatedResult;
-          if (!holdings) return { ...resultObj, updatedResult };
-          addCheckInItems(updatedResult);
-          holdings.slice(0, itemTableLimit).forEach(
-            holding => addHoldingDefinition(holding));
-          return { ...resultObj, updatedResult };
+        const { holdings } = resultObj.result;
+        if (holdings) {
+          addCheckInItems(result);
+          holdings.slice(0, itemTableLimit).forEach((holding) => {
+            addHoldingDefinition(holding);
+            if (numItemsProcessed < 0 && holding.location) {
+              locationCodes.add(holding.location[0].code);
+              numItemsProcessed += 1;
```

Comment:
I think it would be clearer (and more declarative) to separate these things out, so that you don't need to keep a running tally. If I understand correctly, what you've written could also be written:

```
holdings.slice(0, itemTableLimit).forEach((holding) => {addHoldingDefinition(holding);};
holdingsToProcess = holdings.slice(0, itemTableLimit).filter((holding) => {... something...});
holdingsToProcesss.forEach((holding) => {locationCodes.add(holding.location[0].code);});
const numItemsProcessed = holdingsToProcess.length
```

Reply:
Not sure what the `filter` on the second line is for or why we would want to repeat `holdings.slice(0, itemTableLimit)`. I do not find this that much clearer. I think once my error was fixed where I had `0` instead of `itemTableLimit` it is now clearer what the intent is. But let me know if you don't agree.